src/main/java/build/buildfarm/server/WriteStreamObserver.java

```diff
@@ -416,33 +416,29 @@ private void requestNextIfReady() {
   private FeedbackOutputStream getOutput() throws IOException {
     if (out == null) {
       out = write.getOutput(deadlineAfter, deadlineAfterUnits, this::onNewlyReadyRequestNext);
+      if (out != null) {
+        withCancellation.addListener(
+            context -> {
+              try {
+                out.close();
```

Comment:
I found it would repeat to report out is nullpointer here. I made a patch to 1.14.1, the L419 is actually here
```
java.lang.NullPointerException
        at build.buildfarm.server.WriteStreamObserver.lambda$getOutput$0(WriteStreamObserver.java:419)
        at io.grpc.Context$ExecutableListener.run(Context.java:1092)
        at com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:30)
        at io.grpc.Context$ExecutableListener.deliver(Context.java:1084)
        at io.grpc.Context$CancellableContext.notifyAndClearListeners(Context.java:873)
        at io.grpc.Context$CancellableContext.cancel(Context.java:846)
        at io.grpc.Context$CancellableContext.isCancelled(Context.java:917)
        at io.grpc.Context$CancellableContext.cancellationCause(Context.java:925)
        at io.grpc.Context$CancellableContext$1.cancelled(Context.java:758)
        at io.grpc.Context$ExecutableListener.run(Context.java:1092)
        at io.grpc.Context$DirectExecutor.execute(Context.java:1109)
        at io.grpc.Context$ExecutableListener.deliver(Context.java:1084)
        at io.grpc.Context$CancellableContext.notifyAndClearListeners(Context.java:878)
        at io.grpc.Context$CancellableContext.cancel(Context.java:846)
        at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.closedInternal(ServerCallImpl.java:358)
        at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.closed(ServerCallImpl.java:341)
        at io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1Closed.runInContext(ServerImpl.java:844)
        at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

Reply:
Although the cluster is still running and normal to handle all the tasks, it repeat to report this exception every second.

Reply:
Sorry, that was definitely an oversight. I've updated this branch to incorporate a null check for that.